src/ui/frmpreferences.cpp

```diff
@@ -92,10 +92,15 @@ void frmPreferences::on_treeWidget_currentItemChanged(QTreeWidgetItem *current,
 
 void frmPreferences::on_buttonBox_accepted()
 {
-    if(m_keyGrabber->findConflicts()){
+    if(m_keyGrabber->hasConflicts()){
+        //Try our best to show the error to the user immediately.
+        ui->treeWidget->setCurrentItem( ui->treeWidget->topLevelItem(4) );
```

Comment:
It's true that the other code uses indices but the point is that they're not hardcoded, so we maintain a relationship between the treeWidget and the stackedWidget indices. Try this:

    ui->treeWidget->setCurrentItem(ui->treeWidget->topLevelItem(ui->stackedWidget->indexOf(ui->pageShortcuts)));